src/Integration/Binding/ProjectRuleSetWriter.cs

```diff
@@ -87,15 +87,21 @@ private bool TryUpdateExistingProjectRuleSet(string solutionRuleSetPath, string
         /// <summary>
         /// Remove all rule set inclusions which exist under the specified root directory.
         /// </summary>
-        internal /* testing purposes */  static void RemoveAllIncludesUnderRoot(RuleSet ruleSet, string rootDirectory)
+        internal /* testing purposes */  void RemoveAllIncludesUnderRoot(RuleSet ruleSet, string rootDirectory)
         {
             Debug.Assert(!string.IsNullOrWhiteSpace(rootDirectory));
-            
+
+            string ruleSetRoot = PathHelper.ForceDirectoryEnding(Path.GetDirectoryName(ruleSet.FilePath));
+
             // List<T> required as RuleSetIncludes will be modified
             List<RuleSetInclude> toRemove = ruleSet.RuleSetIncludes
                                             .Where(include =>
                                             {
-                                                string fullIncludePath = PathHelper.ResolveRelativePath(include.FilePath, rootDirectory);
+                                                string fullIncludePath = PathHelper.ResolveRelativePath(include.FilePath, ruleSetRoot);
+                                                if (!this.FileSystem.FileExists(fullIncludePath))
+                                                {
+                                                    return false;
+                                                }
```

Comment:
Come to think of it, does it matter whether the file exists or not? If it is under the SonarQube folder then shouldn't we remove it, regardless of whether it exists or not?